File: frontend/types/alert.ts
export type AlertType = "HIGH_WASTE_RATE" | "LONG_FAULT_DURATION" | "LONG_STOP_DURATION";
export type AlertSeverity = "INFO" | "WARNING" | "CRITICAL";

export interface Alert {
  id: number;
  alertType: AlertType;
  severity: AlertSeverity;
  machineId?: number | null;
  machineName?: string | null;
  departmentId?: number | null;
  departmentName?: string | null;
  materialId?: number | null;
  materialTrackingCode?: string | null;
  message: string;
  thresholdValue?: number | null;
  actualValue?: number | null;
  isRead: boolean;
  createdAt: string;
}

File: backend/src/main/java/com/meslite/domain/alert/Alert.java
package com.meslite.domain.alert;

import com.meslite.common.BaseEntity;
import com.meslite.domain.machine.Machine;
import com.meslite.domain.material.Material;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import java.math.BigDecimal;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "alerts")
public class Alert extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Enumerated(EnumType.STRING)
    @Column(name = "alert_type", nullable = false, length = 50)
    private AlertType alertType;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private AlertSeverity severity = AlertSeverity.WARNING;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "machine_id")
    private Machine machine;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "material_id")
    private Material material;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String message;

    @Column(name = "threshold_value", precision = 10, scale = 3)
    private BigDecimal thresholdValue;

    @Column(name = "actual_value", precision = 10, scale = 3)
    private BigDecimal actualValue;

    @Column(name = "is_read", nullable = false)
    private Boolean isRead = false;
}

